reference/constraints/NoSuspiciousCharacters.rst

```diff
@@ -0,0 +1,157 @@
+        }
+
+    .. code-block:: yaml
+
+        # config/validator/validation.yaml
+        App\Entity\User:
+            properties:
+                username:
+                    - NoSuspiciousCharacters: ~
+
+    .. code-block:: xml
+
+        <!-- config/validator/validation.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping https://symfony.com/schema/dic/constraint-mapping/constraint-mapping-1.0.xsd">
+
+            <class name="App\Entity\User">
+                <property name="username">
+                    <constraint name="NoSuspiciousCharacters"/>
+                </property>
+            </class>
+        </constraint-mapping>
+
+    .. code-block:: php
+
+        // src/Entity/User.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+
+        class User
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('username', new Assert\NoSuspiciousCharacters());
+            }
+        }
+
+.. include:: /reference/constraints/_empty-values-are-valid.rst.inc
+
+Options
+-------
+
+``checks``
+~~~~~~~~~~
+
+**type**: ``integer`` **default**: all
+
+This option is a bitmask of the checks you want to perform on the string:
+
+* ``NoSuspiciousCharacters::CHECK_INVISIBLE`` checks for the presence of invisible characters such as zero-width spaces, or character sequences that are likely not to display, such as multiple occurrences of the same non-spacing mark.
+* ``NoSuspiciousCharacters::CHECK_MIXED_NUMBERS`` (since ICU 58) checks for numbers from different numbering systems.
+* ``NoSuspiciousCharacters::CHECK_HIDDEN_OVERLAY`` (since ICU 62) checks for combining characters hidden in their preceding one.
+
+You can also configure additional requirements using :ref:`locales <locales>` and
+:ref:`restrictionLevel <restrictionlevel>`.
+
+``locales``
+~~~~~~~~~~~
+
+**type**: ``array`` **default**: :ref:`framework.enabled_locales <reference-enabled-locales>`
+
+Restrict the string's characters to those normally used with the associated languages.
+
+For example, the character "π" would be considered suspicious if you restricted the
+locale to "English", because the Greek script is not associated with it.
+
+Passing an empty array, or configuring :ref:`restrictionLevel <restrictionlevel>` to
+``NoSuspiciousCharacters::RESTRICTION_LEVEL_NONE`` will disable this requirement.
+
+``restrictionLevel``
+~~~~~~~~~~~~~~~~~~~~
+
+**type**: ``integer`` **default**: ``NoSuspiciousCharacters::RESTRICTION_LEVEL_MODERATE`` on ICU >= 58, else ``NoSuspiciousCharacters::RESTRICTION_LEVEL_SINGLE_SCRIPT``
```

Comment:
```suggestion
**type**: ``integer`` **default**: ``NoSuspiciousCharacters::RESTRICTION_LEVEL_MODERATE`` on ICU >= 58, otherwise ``NoSuspiciousCharacters::RESTRICTION_LEVEL_SINGLE_SCRIPT``
```

Reply:
I recall you already asked this change :sweat_smile: is this because `else` is associated with coding?